modules/fbx_importer/data/fbx_material.h

```diff
@@ -115,39 +111,39 @@ struct FBXMaterial : Reference {
-			{"Maya|emissionColor|file", SpatialMaterial::TextureParam::TEXTURE_EMISSION },
-			{"3dsMax|Parameters|emission_map", SpatialMaterial::TextureParam::TEXTURE_EMISSION },
-			{"Maya|TEX_emissive_map|file", SpatialMaterial::TextureParam::TEXTURE_EMISSION },
-			/* Metallic */
-			{"Maya|metalness|file", SpatialMaterial::TextureParam::TEXTURE_METALLIC },
-			{"3dsMax|Parameters|metalness_map", SpatialMaterial::TextureParam::TEXTURE_METALLIC },
-			{"Maya|TEX_metallic_map|file", SpatialMaterial::TextureParam::TEXTURE_METALLIC },
-			{"SpecularColor", SpatialMaterial::TextureParam::TEXTURE_METALLIC },
-			{"Maya|SpecularTexture", SpatialMaterial::TextureParam::TEXTURE_METALLIC },
-			{"ShininessExponent", SpatialMaterial::TextureParam::TEXTURE_METALLIC },
-            /* Roughness */
-			{"Maya|diffuseRoughness|file", SpatialMaterial::TextureParam::TEXTURE_ROUGHNESS },
-			{"3dsMax|Parameters|roughness_map", SpatialMaterial::TextureParam::TEXTURE_ROUGHNESS },
-			{"Maya|TEX_roughness_map|file", SpatialMaterial::TextureParam::TEXTURE_ROUGHNESS },
-		    /* Normal */
-			{"NormalMap", SpatialMaterial::TextureParam::TEXTURE_NORMAL},
-			{"Bump", SpatialMaterial::TextureParam::TEXTURE_NORMAL},
-			{"3dsMax|Parameters|bump_map", SpatialMaterial::TextureParam::TEXTURE_NORMAL},
-			{"Maya|NormalTexture", SpatialMaterial::TextureParam::TEXTURE_NORMAL},
-			{"Maya|normalCamera|file", SpatialMaterial::TextureParam::TEXTURE_NORMAL},
-			{"Maya|TEX_normal_map|file", SpatialMaterial::TextureParam::TEXTURE_NORMAL},
-			/* AO */
-			{"Maya|TEX_ao_map|file", SpatialMaterial::TextureParam::TEXTURE_AMBIENT_OCCLUSION},
+	const std::map<std::string, SpatialMaterial::TextureParam> fbx_mapping_paths = {
+		/* Diffuse */
+		{ "DiffuseColor", SpatialMaterial::TextureParam::TEXTURE_ALBEDO },
+		{ "Maya|DiffuseTexture", SpatialMaterial::TextureParam::TEXTURE_ALBEDO },
+		{ "Maya|baseColor|file", SpatialMaterial::TextureParam::TEXTURE_ALBEDO },
+		{ "3dsMax|Parameters|base_color_map", SpatialMaterial::TextureParam::TEXTURE_ALBEDO },
+		{ "Maya|TEX_color_map|file", SpatialMaterial::TextureParam::TEXTURE_ALBEDO },
+		/* Emission */
+		{ "EmissiveColor", SpatialMaterial::TextureParam::TEXTURE_EMISSION },
+		{ "EmissiveFactor", SpatialMaterial::TextureParam::TEXTURE_EMISSION },
+		{ "Maya|emissionColor|file", SpatialMaterial::TextureParam::TEXTURE_EMISSION },
+		{ "3dsMax|Parameters|emission_map", SpatialMaterial::TextureParam::TEXTURE_EMISSION },
+		{ "Maya|TEX_emissive_map|file", SpatialMaterial::TextureParam::TEXTURE_EMISSION },
+		/* Metallic */
+		{ "Maya|metalness|file", SpatialMaterial::TextureParam::TEXTURE_METALLIC },
+		{ "3dsMax|Parameters|metalness_map", SpatialMaterial::TextureParam::TEXTURE_METALLIC },
+		{ "Maya|TEX_metallic_map|file", SpatialMaterial::TextureParam::TEXTURE_METALLIC },
+		{ "SpecularColor", SpatialMaterial::TextureParam::TEXTURE_METALLIC },
+		{ "Maya|SpecularTexture", SpatialMaterial::TextureParam::TEXTURE_METALLIC },
```

Comment:
"SpecularColor" and "SpecularTexture" are not metallic textures. They're a combination of base color and roughness. Metallic is glossiness...

Reply:
See https://github.com/godotengine/godot/pull/38438/files#diff-de1d3a22612435f4a0efc95d83e02952R1578

Reply:
these will be removed, the conversion here for each type of material type is not going to be added since there are 40-60 different properties to support

Reply:
I believe SpecularColor, SpecularTexture is for the legacy texturing of FBX. As per our discussion, you mean that legacy materials will be removed? That makes sense, but you probably want to go into depth.

As per the discussion we mentioned there's legacy texturing and like 3+ sets of PBR texturing stacked.

Reply:
no, just incompatible mappings which require conversion, so some legacy stuff will remain; the point will be to keep compatibility for the basics (like color), specular is not something really that matters a lot in the grand scheme of things to getting this over the line for me, I just want to get this sorted and merged, then we can add whatever and then worry about all the finer detail conversion for the materials.

At best materials won't really produce the same results in different packages for me, so converting them seems pointless, my intention is to use these as placeholders for now, and actually spend time ensuring the mappings from maya are compatible, because IMHO they're not all going to be the same formula if they're called specular, they could have an FBX specific way of handling this in one system like GLTF2 but it seems like a guess to assume it's the same algorithm for specular, especially since FBX seems to be built on top of their own everything